runner.go

```diff
@@ -1,128 +1,188 @@
-	EnvironmentID int      `json:"environment_id,omitempty"`
+// startRun starts a new Rainforest run & depending on passed flags monitors its execution
+func startRun(c *cli.Context) error {
+	params, err := makeRunParams(c)
+	if err != nil {
+		return cli.NewExitError(err.Error(), 1)
+	}
+
+	runStatus, err := api.CreateRun(params)
+	if err != nil {
+		return cli.NewExitError(err.Error(), 1)
+	}
+	fmt.Printf("Run %v has been created...\n", runStatus.ID)
+
+	// if foreground flag is enabled we'll monitor run status
+	if c.Bool("fg") {
+		// Create two channels to communicate with the polling goroutine
+		// One that will tick when it's time to poll and the other to gather final state
+		t := time.NewTicker(runStatusPollInterval)
+		statusChan := make(chan statusWithError, 0)
+		go updateRunStatus(c, runStatus.ID, t, statusChan)
+
+		// This channel readout will block until updateRunStatus pushed final result to it
+		finalState := <-statusChan
+		if finalState.err != nil {
+			return cli.NewExitError(finalState.err.Error(), 1)
+		}
+
+		if finalState.status.FrontendURL != "" {
+			fmt.Printf("The detailed results are available at %v\n", finalState.status.FrontendURL)
+		}
+
+		if finalState.status.Result != "passed" {
+			return cli.NewExitError("", 1)
+		}
+
+		return nil
+	}
+
+	return nil
 }
 
-type runResponse struct {
-	ID           int    `json:"id"`
-	State        string `json:"state"`
-	StateDetails struct {
-		Name         string `json:"name"`
-		IsFinalState bool   `json:"is_final_state"`
-	} `json:"state_details"`
-	Result          string `json:"result"`
-	CurrentProgress struct {
-		Percent  int `json:"percent"`
-		Total    int `json:"total"`
-		Complete int `json:"complete"`
-		NoResult int `json:"no_result"`
-	} `json:"current_progress"`
-	FrontendURL string `json:"frontend_url,omitempty"`
+// statusWithError is a helper type used to send RunStatus and error using single channel
+type statusWithError struct {
+	status rainforest.RunStatus
+	err    error
 }
 
-func createRun() {
-	params := makeParams()
-	resBody := postRun(params)
+func updateRunStatus(c *cli.Context, runID int, t *time.Ticker, resChan chan statusWithError) {
+	for {
+		// Wait for tick
+		<-t.C
+		newStatus, err := api.CheckRunStatus(runID)
+		if err != nil {
+			resChan <- statusWithError{status: newStatus, err: err}
+		}
 
-	fmt.Println(resBody)
-}
+		isFinalState := newStatus.StateDetails.IsFinalState
+		state := newStatus.State
+		currentPercent := newStatus.CurrentProgress.Percent
 
-func makeParams() *runParams {
-	if testIDs != "" {
-		testIDsSlice := stringToIntSlice(testIDs)
-		return &runParams{Tests: testIDsSlice}
-	}
-	tags = strings.TrimSpace(tags)
-	var slicedTags []string
-	if tags != "" {
-		slicedTags = strings.Split(tags, ",")
-		for i, slice := range slicedTags {
-			slicedTags[i] = strings.TrimSpace(slice)
+		if !isFinalState {
+			fmt.Printf("Run %v is %v and is %v%% complete\n", runID, state, currentPercent)
```

Comment:
can you use ``log`` here not ``fmt``? it'll output a nice parsable log with time